timeline-chart/src/components/time-graph-annotation.ts

```diff
@@ -38,16 +39,26 @@ export class TimeGraphAnnotationComponent extends TimeGraphComponent {
     update(opts: TimeGraphAnnotationComponentOptions, style?: TimeGraphAnnotationStyle): void {
         if (opts) {
             this._options.position.x = opts.position.x;
-            this._options.position.y = this._row.position.y + (this._row.height * 0.5);
+            this.updateYPosition();
         }
         if (style) {
             this._style.color = style.color;
             this._style.size = style.size;
-            this._style.symbol = style.symbol;
+            this._style.verticalAlign = style.verticalAlign;
+            this.updateYPosition();
         }
         super.update();
     }
 
+    private updateYPosition() {
+        const align = this._style.verticalAlign;
+        const size = this._style.size;
+        if (!!size) {
+            const offset = align == 'top' ? size / 2 : align == 'bottom' ? this._row.height - size / 2 : this._row.height;
```

Comment:
The symbol size is already half the height of the symbol (e.g. radius). So for 'top' it should be size, for 'bottom' it should be height - size, and for 'middle' it should be height / 2?

Reply:
Let's try with numbers:

size = 7, the row and the symbol are size 14.

top = 7
Bottom = 14-7 = 7
center = 14/2 = 7

I went for size/2 so we can be clamped to the limit

But I am not sure.

Reply:
In your example if row and symbol are 14 (meaning full-height symbol) then its size is 7 and the center y should also be 7 for all cases. But the code gives: 3.5, 11.5 and 14.

For half-height symbol with row height 20, the size is 5, we would want: 5, 20-5=15, 20/2=10.

Reply:
I am not really convinced here. look at this:

top:
![image](https://user-images.githubusercontent.com/419535/110133406-47532900-7d9a-11eb-833e-d7c0e2777622.png)
center:
![image](https://user-images.githubusercontent.com/419535/110133416-491cec80-7d9a-11eb-9505-54da38e60f1a.png)
bottom:
![image](https://user-images.githubusercontent.com/419535/110133424-4a4e1980-7d9a-11eb-983a-fc26ea518d61.png)


vs

top:
![image](https://user-images.githubusercontent.com/419535/110133936-deb87c00-7d9a-11eb-9c45-db1008c4f8d6.png)
center:
![image](https://user-images.githubusercontent.com/419535/110133430-4b7f4680-7d9a-11eb-9b3f-eb6e01f3fb2d.png)
bottom:
![image](https://user-images.githubusercontent.com/419535/110133722-a1ec8500-7d9a-11eb-9b5f-28320c91b24d.png)